Broken disabled HDF5 builds

The latest PR was missing some `#ifdef`'s to keep non-HDF5 builds working. when compiling with `PARTHENON_DISABLE_HDF5=ON` I get this error while building:
```
/home/scratch.fglines_gpu/projects/ats-5-benchmarks/parthenon/src/outputs/restart_hdf5.cpp: In member function 'virtual parthenon::RestartReader::MeshInfo parthenon::RestartReaderHDF5::GetMeshInfo() const':
/home/scratch.fglines_gpu/projects/ats-5-benchmarks/parthenon/src/outputs/restart_hdf5.cpp:138:38: error: 'fh_' was not declared in this scope
  138 |       PARTHENON_HDF5_CHECK(H5Lexists(fh_, "Blocks/derefinement_count", H5P_DEFAULT));
      |                                      ^~~
/home/scratch.fglines_gpu/projects/ats-5-benchmarks/parthenon/src/outputs/restart_hdf5.cpp:138:72: error: 'H5P_DEFAULT' was not declared in this scope; did you mean 'RTLD_DEFAULT'?
  138 |       PARTHENON_HDF5_CHECK(H5Lexists(fh_, "Blocks/derefinement_count", H5P_DEFAULT));
      |                                                                        ^~~~~~~~~~~
      |                                                                        RTLD_DEFAULT
/home/scratch.fglines_gpu/projects/ats-5-benchmarks/parthenon/src/outputs/restart_hdf5.cpp:138:28: error: 'H5Lexists' was not declared in this scope
  138 |       PARTHENON_HDF5_CHECK(H5Lexists(fh_, "Blocks/derefinement_count", H5P_DEFAULT));
      |                            ^~~~~~~~~
/home/scratch.fglines_gpu/projects/ats-5-benchmarks/parthenon/src/outputs/restart_hdf5.cpp:138:7: error: 'PARTHENON_HDF5_CHECK' was not declared in this scope; did you mean 'PARTHENON_MPI_CHECK'?
  138 |       PARTHENON_HDF5_CHECK(H5Lexists(fh_, "Blocks/derefinement_count", H5P_DEFAULT));
      |       ^~~~~~~~~~~~~~~~~~~~
      |       PARTHENON_MPI_CHECK
 ```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,7 @@
 - [[PR 1004]](https://github.com/parthenon-hpc-lab/parthenon/pull/1004) Allow parameter modification from an input file for restarts
 
 ### Fixed (not changing behavior/API/variables/...)
+- [[PR 1095]](https://github.com/parthenon-hpc-lab/parthenon/pull/1095) Add missing include guards in hdf5 restart
 - [[PR 1093]](https://github.com/parthenon-hpc-lab/parthenon/pull/1093) Fix forest size for symmetry dimensions
 - [[PR 1089]](https://github.com/parthenon-hpc-lab/parthenon/pull/1089) Fix loading restart files without derefinement counter
 - [[PR 1079]](https://github.com/parthenon-hpc-lab/parthenon/pull/1079) Address XDMF/Visit Issues

--- a/src/outputs/restart_hdf5.cpp
+++ b/src/outputs/restart_hdf5.cpp
@@ -119,7 +119,7 @@ RestartReaderHDF5::MeshInfo RestartReaderHDF5::GetMeshInfo() const {
 #ifndef ENABLE_HDF5
   PARTHENON_FAIL("Restart functionality is not available because HDF5 is disabled");
 #else
-  RestartReaderHDF5::MeshInfo mesh_info;  
+  RestartReaderHDF5::MeshInfo mesh_info;
   mesh_info.nbnew = GetAttr<int>("Info", "NBNew");
   mesh_info.nbdel = GetAttr<int>("Info", "NBDel");
   mesh_info.nbtotal = GetAttr<int>("Info", "NumMeshBlocks");

--- a/src/outputs/restart_hdf5.cpp
+++ b/src/outputs/restart_hdf5.cpp
@@ -116,7 +116,10 @@ RestartReaderHDF5::SparseInfo RestartReaderHDF5::GetSparseInfo() const {
 }
 
 RestartReaderHDF5::MeshInfo RestartReaderHDF5::GetMeshInfo() const {
-  RestartReaderHDF5::MeshInfo mesh_info;
+#ifndef ENABLE_HDF5
+  PARTHENON_FAIL("Restart functionality is not available because HDF5 is disabled");
+#else
+  RestartReaderHDF5::MeshInfo mesh_info;  
   mesh_info.nbnew = GetAttr<int>("Info", "NBNew");
   mesh_info.nbdel = GetAttr<int>("Info", "NBDel");
   mesh_info.nbtotal = GetAttr<int>("Info", "NumMeshBlocks");
@@ -149,6 +152,7 @@ RestartReaderHDF5::MeshInfo RestartReaderHDF5::GetMeshInfo() const {
     mesh_info.derefinement_count = std::vector<int>(mesh_info.nbtotal, 0);
   }
   return mesh_info;
+#endif
 }
 
 SimTime RestartReaderHDF5::GetTimeInfo() const {

